contracts/smart-contract-wallet/SmartAccount.sol

```diff
@@ -344,7 +353,7 @@ contract SmartAccount is
                     // The signature data for contract signatures is appended to the concatenated signatures and the offset is stored in s
                     contractSignature := add(add(signatures, s), 0x20)
                 }
-                require(ISignatureValidator(_signer).isValidSignature(dataHash, contractSignature) == EIP1271_MAGIC_VALUE, "BSA024");
+                if(ISignatureValidator(_signer).isValidSignature(dataHash, contractSignature) != EIP1271_MAGIC_VALUE) revert WrongContractSignature(contractSignature);
```

Comment:
taking a note so it helps with resolving conflicts
we should probably write test cases that catches these error code in foundry test contract and in off chain test script.

Reply:
what is the purpose of such a test case? is it to test how can we catch it? or to test gas consumption?

Reply:
catching..